common-util/src/main/java/org/sunbird/services/sso/impl/KeyCloakServiceImpl.java

```diff
@@ -478,11 +489,11 @@ public String getEmailVerifiedUpdatedFlag(String userId) {
   private UserRepresentation createUserReqObj(Map<String, Object> request) {
     CredentialRepresentation credential = new CredentialRepresentation();
     UserRepresentation user = new UserRepresentation();
-    if (isNotNull(request.get(JsonKey.PROVIDER))) {
+    if (isNotNull(request.get(JsonKey.CHANNEL))) {
```

Comment:
Use instead loginId which is available for existing users (userName@provider) and new users (userName@channel) in sync.